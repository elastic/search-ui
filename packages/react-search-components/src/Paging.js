import PropTypes from "prop-types";
import React from "react";
import RCPagination from "rc-pagination";

function Paging({ current, resultsPerPage, onChange, totalPages }) {
  return (
    <RCPagination
      current={current}
      onChange={onChange}
      pageSize={resultsPerPage}
      total={totalPages * resultsPerPage}
      className="sui-paging"
    />
  );
}

Paging.propTypes = {
  current: PropTypes.number.isRequired,
  onChange: PropTypes.func.isRequired,
  resultsPerPage: PropTypes.number.isRequired,
  totalPages: PropTypes.number.isRequired
};

export default Paging;
