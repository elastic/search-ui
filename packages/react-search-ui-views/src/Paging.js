import PropTypes from "prop-types";
import React from "react";
import RCPagination from "rc-pagination";
import enUsLocale from "rc-pagination/lib/locale/en_US";

import { appendClassName } from "./view-helpers";

function Paging({
  className,
  current,
  resultsPerPage,
  onChange,
  totalPages,
  ...rest
}) {
  return (
    <RCPagination
      current={current}
      onChange={onChange}
      pageSize={resultsPerPage}
      total={totalPages * resultsPerPage}
      className={appendClassName("sui-paging", className)}
      locale={enUsLocale}
      {...rest}
    />
  );
}

Paging.propTypes = {
  current: PropTypes.number.isRequired,
  onChange: PropTypes.func.isRequired,
  resultsPerPage: PropTypes.number.isRequired,
  totalPages: PropTypes.number.isRequired,
  className: PropTypes.string
};

export default Paging;
