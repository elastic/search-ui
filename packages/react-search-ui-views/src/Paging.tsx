import React from "react";
import RCPagination from "rc-pagination";
import enUsLocale from "rc-pagination/lib/locale/en_US";

import { appendClassName } from "./view-helpers";
import type { SearchContextState } from "@elastic/search-ui";
import { Rename, BaseContainerProps } from "./types";

export type PagingContainerContext = Pick<
  SearchContextState,
  "current" | "resultsPerPage" | "totalPages" | "setCurrent"
>;

export type PagingViewProps = Rename<
  BaseContainerProps & PagingContainerContext,
  {
    setCurrent: "onChange";
  }
>;

export type PagingContainerProps = BaseContainerProps &
  PagingContainerContext & {
    view?: React.ComponentType<PagingViewProps>;
  };

function Paging({
  className,
  current,
  resultsPerPage,
  onChange,
  totalPages,
  ...rest
}: PagingViewProps) {
  return (
    <RCPagination
      current={current}
      onChange={onChange}
      pageSize={resultsPerPage}
      total={totalPages * resultsPerPage}
      className={appendClassName("sui-paging", className)}
      locale={enUsLocale}
      {...rest}
    />
  );
}

export default Paging;
