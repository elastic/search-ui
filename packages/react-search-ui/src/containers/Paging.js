import PropTypes from "prop-types";

import { withSearch } from "..";
import { Paging } from "@elastic/react-search-ui-views";

export function PagingContainer({
  className,
  current,
  resultsPerPage,
  setCurrent,
  totalPages,
  view,
  ...rest
}) {
  if (totalPages === 0) return null;

  const View = view || Paging;

  return View({
    className,
    current,
    resultsPerPage,
    totalPages,
    onChange: setCurrent,
    ...rest
  });
}

PagingContainer.propTypes = {
  // Props
  className: PropTypes.string,
  view: PropTypes.func,
  // State
  current: PropTypes.number.isRequired,
  resultsPerPage: PropTypes.number.isRequired,
  totalPages: PropTypes.number.isRequired,
  // Action
  setCurrent: PropTypes.func.isRequired
};

export default withSearch(
  ({ current, resultsPerPage, totalPages, setCurrent }) => ({
    current,
    resultsPerPage,
    totalPages,
    setCurrent
  })
)(PagingContainer);
