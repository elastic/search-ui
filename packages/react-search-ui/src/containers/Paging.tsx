import React from "react";

import { withSearch } from "..";
import {
  Paging,
  PagingContainerContext,
  PagingContainerProps,
  PagingViewProps
} from "@elastic/react-search-ui-views";

export function PagingContainer({
  className,
  current,
  resultsPerPage,
  setCurrent,
  totalPages,
  view,
  ...rest
}: PagingContainerProps) {
  if (totalPages === 0) return null;

  const View: React.ComponentType<PagingViewProps> = view || Paging;

  const viewProps = {
    className,
    current,
    resultsPerPage,
    totalPages,
    onChange: setCurrent,
    ...rest
  };

  return <View {...viewProps} />;
}

export default withSearch<PagingContainerProps, PagingContainerContext>(
  ({ current, resultsPerPage, totalPages, setCurrent }) => ({
    current,
    resultsPerPage,
    totalPages,
    setCurrent
  })
)(PagingContainer);
