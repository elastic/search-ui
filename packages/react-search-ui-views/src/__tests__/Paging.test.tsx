import React from "react";
import Paging from "../Paging";
import { shallow } from "enzyme";
import { PagingViewProps } from "../Paging";

const params = {
  current: 1,
  onChange: () => ({}),
  resultsPerPage: 10,
  totalPages: 100
};

it("renders correctly", () => {
  const wrapper = shallow(<Paging {...params} />);
  expect(wrapper).toMatchSnapshot();
});

it("renders with className prop applied", () => {
  const customClassName = "test-class";
  const wrapper = shallow<typeof Paging>(
    <Paging className={customClassName} {...params} />
  );
  const { className } = wrapper.dive().props() as PagingViewProps;
  expect(className).toEqual("rc-pagination sui-paging test-class");
});
