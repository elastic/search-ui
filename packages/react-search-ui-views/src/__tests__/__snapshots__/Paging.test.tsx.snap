// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Pagination
  className="sui-paging"
  current={1}
  defaultCurrent={1}
  defaultPageSize={10}
  hideOnSinglePage={false}
  itemRender={[Function]}
  locale={
    Object {
      "items_per_page": "/ page",
      "jump_to": "Go to",
      "jump_to_confirm": "confirm",
      "next_3": "Next 3 Pages",
      "next_5": "Next 5 Pages",
      "next_page": "Next Page",
      "page": "",
      "prev_3": "Previous 3 Pages",
      "prev_5": "Previous 5 Pages",
      "prev_page": "Previous Page",
    }
  }
  onChange={[Function]}
  onShowSizeChange={[Function]}
  pageSize={10}
  prefixCls="rc-pagination"
  selectComponentClass={null}
  selectPrefixCls="rc-select"
  showLessItems={false}
  showPrevNextJumpers={true}
  showQuickJumper={false}
  showSizeChanger={false}
  showTitle={true}
  style={Object {}}
  total={1000}
/>
`;
